### PR TITLE
Notifies user for invalid fields while validating

### LINKED ISSUE #993: Provide validation hints for common mistakes?
# Overview

For example, one can define `missingValues` on the resource level instead of the schema level. It's a common mistake and according to the standards a resource having `missingValues` is perfectly valid (because props set are customizable and open-ended). In this case (and similar), the framework might emit a warning in the validation report.

---

Missed schema level (`resource -> fields` instead of `resource -> schema -> fields`) - https://github.com/frictionlessdata/frictionless-py/issues/1043

### PR DESCRIPTION
**Pull request by shashigharti**
* Added check for invalid fields: missingValues and fields (in resource descriptor)
* Added check for invalid field: missingValues (in package descriptor)
* Added tests
* Fixed old tests

- fixes #993

### CODE REVIEW

**Comment by shashigharti** [comment]
@roll This is ready for the review.

I also tried adding check for the following in package.json. It quietly passes the validation but doesn't trigger metadata_validate function. What would be your suggestion to catch this error?

```
{
  "name": "test-tabulator",
  "fields": [
          {
            "name": "id",
            "type": "integer",            
          },
          {
            "name": "name",
            "type": "string"
          }
        ]
}
```
screenshot-localhost_8888-2022 05 31-17_23_34


Thank you!

File: frictionless/package/package.py [hunk 1]
```diff
@@ -704,6 +704,13 @@ def metadata_process(self):
                 dict.__setitem__(self, "resources", resources)
 
     def metadata_validate(self):
+        # Check invalid properties
+        invalid_fields = {"missingValues": "schema/fields"}
+        for invalid_field, resource in invalid_fields.items():
+            if invalid_field in self:
+                note = f"{invalid_field} should be part of {resource} not package."
+                error = errors.ResourceError(note=note)
+                raise FrictionlessException(error)
```
Comment (roll) [inline frictionless/package/package.py]: You need to `yield` errors here as in other places in `metadata_validate` functions (the same for resource)

Comment (roll) [reply frictionless/package/package.py]: You can write a test for it:

```
package = Package('with-bad-props')
package.metadata_errors == [...this error...]
```

File: tests/package/validate/test_general.py [hunk 2]
```diff
@@ -345,3 +345,14 @@ def test_validate_package_with_resource_data_is_a_string_issue_977():
...
     ]
+
+
+def test_validate_package_with_missing_values_993():
+    package = Package(descriptor="data/package-with-missingvalues.json")
+    report = package.validate()
+    assert report.flatten(["code", "message"]) == [
+        [
+            "resource-error",
+            "The data resource has an error: missingValues should be part of schema/fields not package.",
```
Comment (roll) [inline tests/package/validate/test_general.py]: "missingValues" should be set as "resource.schema.missingValues" (not "resource.missingValues")

File: tests/resource/validate/test_general.py [hunk 3]
```diff
@@ -519,3 +519,25 @@ def test_validate_resource_duplicate_labels_with_sync_schema_issue_910():
...
+    ]
+
+
+def test_validate_resource_fields_without_schema_993():
+    resource = Resource("data/resource-fields-without-schema.json")
+    report = resource.validate()
+    assert report.flatten(["code", "message"]) == [
+        [
+            "resource-error",
+            "The data resource has an error: fields should be part of schema not resource.",
```
Comment (roll) [inline tests/resource/validate/test_general.py]: "fields" should be set as "resource.schema.fields" (not "resource.fields")

File: frictionless/package/package.py [hunk 4]
```diff
@@ -704,6 +704,13 @@ def metadata_process(self):
                 dict.__setitem__(self, "resources", resources)
 
     def metadata_validate(self):
+        # Check invalid properties
+        invalid_fields = {"missingValues": "schema/fields"}
```
Comment (roll) [inline frictionless/package/package.py]: Regarding your question - shall we check for `fields` here also to catch `package.fields`?

**Review by roll — CHANGES_REQUESTED** [review]
Thanks @shashigharti!

I've added a few comments

**Comment by shashigharti** [comment]
Thank you @roll!.  I have made the changes and it is ready for the review.

File: tests/package/validate/test_general.py [hunk 5]
```diff
@@ -345,3 +345,33 @@ def test_validate_package_with_resource_data_is_a_string_issue_977():
...
+        [
+            "resource-error",
+            '"missingValues" should be set as "resource.schema.missingValues" (not "package.missingValues").',
+        ]
+    ]
+
+
+def test_validate_package_with_fields_993():
+    package = Package(descriptor="data/package-with-fields-993.json")
+    output = list(
```
Comment (roll) [inline tests/package/validate/test_general.py]: Can we just write assertions like this:
```
assert package.metadata_errors[0].code == ...
assert package.metadata_errors[0].note == ...
```
Your version is smart but really makes the reader think while the simpler one doesn't require any additional energy

File: tests/package/validate/test_general.py [hunk 6]
```diff
@@ -345,3 +345,33 @@ def test_validate_package_with_resource_data_is_a_string_issue_977():
     assert report.flatten() == [
         [None, None, None, "package-error"],
     ]
+
+
+def test_validate_package_with_missing_values_993():
```
Comment (roll) [inline tests/package/validate/test_general.py]: I would add also `package.validate` tests

File: tests/resource/validate/test_general.py [hunk 7]
```diff
@@ -519,3 +519,25 @@ def test_validate_resource_duplicate_labels_with_sync_schema_issue_910():
             'Duplicate labels in header is not supported with "schema_sync"',
         ],
     ]
+
+
+def test_validate_resource_with_missing_values_993():
```
Comment (roll) [inline tests/resource/validate/test_general.py]: I would also add `resource.metadata_errors` tests (so it will be 4 for package and 4 for resource)

**Review by roll — CHANGES_REQUESTED** [review]
@shashigharti Nice!

Just a few minor suggestions

File: tests/package/validate/test_general.py (hunk 5 above)
Comment (shashigharti) [reply tests/package/validate/test_general.py]: Thank you for pointing out. I was making it complex unnecessarily :).

**Comment by shashigharti** [comment]
@roll  Thank you a lot! I have made the changes.

**Review by roll — APPROVED** [review]
Perfect!

**Pull request merged**